runtime-transaction/src/compute_budget_program_id_filter.rs

```diff
@@ -1,37 +1,114 @@
 // static account keys has max
 use {
     agave_transaction_view::static_account_keys_frame::MAX_STATIC_ACCOUNTS_PER_PACKET as FILTER_SIZE,
-    solana_builtins_default_costs::MAYBE_BUILTIN_KEY, solana_sdk::pubkey::Pubkey,
+    solana_builtins_default_costs::{BUILTIN_INSTRUCTION_COSTS, MAYBE_BUILTIN_KEY},
+    solana_sdk::pubkey::Pubkey,
 };
 
+#[derive(Default, PartialEq)]
+enum BuiltinCheckStatus {
+    #[default]
+    Unchecked,
+    NotBuiltin,
+    Builtin {
+        is_compute_budget: bool,
+    },
+}
```

Comment:
How about we keep using an `Option` in the filter array but instead of `bool` we use `ProgramKind` which is something like:

```rust
enum ProgramKind {
  NotBuiltin,
  Builtin {
    is_compute_budget: bool,
  }
}
```